src/resourceschemehandler.cc

```diff
@@ -12,7 +12,12 @@ void ResourceSchemeHandler::requestStarted( QWebEngineUrlRequestJob * requestJob
   const QMimeType mineType                    = db.mimeTypeForUrl( url );
   const sptr< Dictionary::DataRequest > reply = this->mManager.getResource( url, content_type );
   content_type                                = mineType.name();
-  if ( reply->isFinished() ) {
+
+  if ( reply == nullptr ) {
```

Comment:
reply==nullptr or   reply.get()==nullptr or !reply?
which one is correct?

Reply:
This is why everyone hates C++ :) 

Based on my current understanding.

1.) `shared_ptr::get()` is an antipattern to temporally escapes shared_ptr's machinery (downgrade to raw pointer). It exists to accommodate existing code and other people's code. In a prefect world, this does not exist (like in C++'s competitors).
2.) `!shared_ptr` is an accident that `0/null` is almost equivalent to `false`. I consider this is a hole in the type system that exists mostly for historical reasons.
3.) Using smart pointers just like a pointer is the intended way of using it. Thus, I consider `reply == nullptr` is the best.

However, based on the standard, 2.) and 3.) are equivalent. `reply == nullptr` will be translated into `!reply`, but I consider the former being less esoteric (controversy!, some people consider `!pointer` is short and easier to read).

https://eel.is/c++draft/util.sharedptr#util.smartptr.shared.cmp

![image](https://github.com/xiaoyifang/goldendict-ng/assets/20123683/85fdfc1e-fe33-4328-b26c-d6a85023efb1)

Reply:
From msvc's STL

https://github.com/microsoft/STL/blob/191d51bb56230069db531f94a02fe0987bf4db85/stl/inc/memory#L1769-L1771

https://github.com/microsoft/STL/blob/191d51bb56230069db531f94a02fe0987bf4db85/stl/inc/memory#L1899-L1902

All 3 options will be compiled into `reply.get()==nullptr` :sweat_smile: 

![image](https://github.com/xiaoyifang/goldendict-ng/assets/20123683/cb7027ab-3b5a-4425-9b95-981e22fd7161)